### PR TITLE
Adds Option to define if a cookie should be set on login.

### PR DESCRIPTION
**Pull request by henrikwirth**
So the idea is to give an option to enable this behaviour.

One option would be to be able to define something in wp-config.php like so:

```
define( 'GRAPHQL_JWT_AUTH_SET_COOKIES', true );
```

Then the `authenticate_user` function would use `wp_signon` with cookies set to true instead of `wp_authenticate`.

I for example need the cookie to be set, so I can use it for the protected Downloads in WooCommerce.

Could be related to this issue: #73 
We could use this approach to also set the RefreshToken as a HttpOnly cookie. Then you could decide to use the JWT in a HttpOnly environment if you need/want to, or still just take it from the response and handle it yourself.

Open for suggestions on how to approach this though. This one works fine for me so far.

### CODE REVIEW

**Comment by henrikwirth** [comment]
Maybe this would rather make sense as a boolean filter in the registration/login mutation?

File: src/Auth.php [hunk 1]
```diff
@@ -355,10 +355,23 @@ public static function is_refresh_token() {
...
+			];
+
+			/**
+			 * Try to authenticate the user with the passed credentials, log him in and set cookies
+			 */
+			$user = wp_signon( $credentials, true );
+		} else {
+			/**
+			 * Try to authenticate the user with the passed credentials
+			 */
```
Comment (renatonascalves) [inline src/Auth.php]:
```suggestion
			 // Try to authenticate the user with the passed credentials
```

File: src/Auth.php [hunk 2]
```diff
@@ -355,10 +355,23 @@ public static function is_refresh_token() {
...
+		if (defined( 'GRAPHQL_JWT_AUTH_SET_COOKIES' ) && ! empty( GRAPHQL_JWT_AUTH_SET_COOKIES ) && GRAPHQL_JWT_AUTH_SET_COOKIES) {
+			$credentials = [
+				'user_login'  => sanitize_user( $username ),
+				'user_password'  => trim( $password ),
+				'remember'  => false,
+			];
+
+			/**
+			 * Try to authenticate the user with the passed credentials, log him in and set cookies
+			 */
```
Comment (renatonascalves) [inline src/Auth.php]:
```suggestion
			 // Try to authenticate the user with the passed credentials, log him in and set cookies
```

**Review by victors1681 — APPROVED** [review]
I did that change before noticed this PR..  looking good.

File: src/Auth.php [hunk 3]
```diff
@@ -353,10 +353,19 @@ public static function is_refresh_token() {
 	 */
 	protected static function authenticate_user( $username, $password ) {
 
-		/**
-		 * Try to authenticate the user with the passed credentials
-		 */
-		$user = wp_authenticate( sanitize_user( $username ), trim( $password ) );
+		if (defined( 'GRAPHQL_JWT_AUTH_SET_COOKIES' ) && ! empty( GRAPHQL_JWT_AUTH_SET_COOKIES ) && GRAPHQL_JWT_AUTH_SET_COOKIES) {
```
Comment (jasonbahl) [inline src/Auth.php:356]:
```suggestion
		if ( defined( 'GRAPHQL_JWT_AUTH_SET_COOKIES' ) && ! empty( GRAPHQL_JWT_AUTH_SET_COOKIES ) && GRAPHQL_JWT_AUTH_SET_COOKIES ) {
```

**Pull request merged**